packages/codegen/src/generateEntityCodegenFile.ts

```diff
@@ -237,6 +239,12 @@ export function generateEntityCodegenFile(config: Config, meta: EntityDbMetadata
         return ${loadLens}(this as any as ${entityName}, fn);
       }
 
+      hasOneThrough<U extends ${EntitySym}, N extends undefined | never, V extends U | N>(
```

Comment:
Do we really want this added to the api of every entity?  Seems like just using the imported `hasOneThrough` would be sufficient.

Reply:
Yeah, good question; without this, the caller did:

```
reaodnly author: Reference<...> = hasOneThrough(this as BookReview, ...);
```

And the `as BookReview` was killing me that it can't be inferred correctly.

Adding "well, for some reason `this` inference is weird" methods to the codebase class is fairly common at this point, but I agree it is not great. Would also be nice to have an in-joist-but-not-the-generated-code comment about this rationale.